### DISCUSSION TITLE
Issue with SpatieMediaLibrary and Spatie's ImageColumn

### DISCUSSION THREAD

**Discussion by PepijnVDK1** [filamentphp/filament#3316] [Help]
Hi, 

I have been trying to have a thumbnail in my listing using the SpatieMediaLibrary and the ImageColumn that comes with.
The file path is not correctly filled in in the thumbnail's source, instead the id is shown. Ex: <img src="2" …

**In my model I have this:**

```
class Portfolio extends Model implements Sortable, HasMedia
{
    use InteractsWithMedia;
    use HasFactory;
    use SortableTrait;
…
```

**In my resource I have this:**

``` 
 return $table
            ->columns([
                SpatieMediaLibraryImageColumn::make('photo')->conversion('thumb'),
```

I have no errors whatssoever.
I know this question doesn't contain much context but I don't have more than this.

Can someone help out?
Thanks in advance!

**Comment by danharrin** [comment]
What happens if you remove the conversion?

**Reply by PepijnVDK1** [reply]
Same thing applies :/

**Reply by danharrin** [reply]
I'm not sure then, try creating a reproduction repo and I will have a look. Make sure to include a seeder that creates all the necessary files and database entires.

**Reply by PepijnVDK1** [reply]
I worked around it by:

- Creating my own ImageColumn
- Extending the SpatieImageColumn
- Overwriting the getImagePath() function and returning the correct thumbnail url

This is probably bad practice but does float my boat.

Thank you though for the effort! Really appreciate it

**Reply by danharrin** [reply]
Very weird, what did you change about `getImagePath()` that made it work? You could PR a fix for this

**Reply by PepijnVDK1** [reply]
```
        $state = $this->getState();

        if ($state) {
            return $state;
        }
```

This is a piece of code you find in the original SpatieMediaLibraryImageColumn getImagePath() function

In my case it returns the id. So I used that ID to get the correct photo and return it:
```
        $state = $this->getState();

        if ($state) {
            $file = MediaLibraryItem::find($state)->getItem();
            return $file->getUrl('thumb');
        }
```
It could also be me doing something wrong which results in this behaviour

**Reply by danharrin** [reply]
Oh wait... do you have a column name on your table with the name `photo`? Or a model accessor?

**Reply by PepijnVDK1** [reply]
A column name with the name photo it is

**Reply by danharrin** [reply]
You don't need that? Thats the whole point of media library? It stores your files in a relationship

So you should drop that column and then you wont need to override anything

**Reply by PepijnVDK1** [reply]
Ahaa, ok, I will check out how it works with the SpatieMediaLibraryManager Pro, it looks like that MediaPicker needs such a column

thanks!

**Reply by danharrin** [reply]
If you still need the column in your database, change the name of the column in Filament and it should still work

Because Filament will always take the value from the database first, and then try to generate a URL if nothing is in the database

```php
SpatieMediaLibraryImageColumn::make('media')->conversion('thumb'),
```
